packages/expo-background-task/android/src/main/java/expo/modules/backgroundtask/BackgroundTaskConsumer.kt

```diff
@@ -0,0 +1,119 @@
+package expo.modules.backgroundtask
+
+import android.content.Context
+import android.util.Log
+import com.facebook.react.common.build.ReactBuildConfig
+import expo.modules.core.interfaces.LifecycleEventListener
+import expo.modules.interfaces.taskManager.TaskConsumer
+import expo.modules.interfaces.taskManager.TaskConsumerInterface
+import expo.modules.interfaces.taskManager.TaskExecutionCallback
+import expo.modules.interfaces.taskManager.TaskInterface
+import expo.modules.interfaces.taskManager.TaskManagerUtilsInterface
+import expo.modules.interfaces.taskManager.TaskServiceProviderHelper
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+class BackgroundTaskConsumer(context: Context?, taskManagerUtils: TaskManagerUtilsInterface?) :
+  TaskConsumer(context, taskManagerUtils), TaskConsumerInterface, LifecycleEventListener {
+  companion object {
+    const val BACKGROUND_TASK_TYPE: String = "expo-background-task"
+    const val DEFAULT_INTERVAL_MINUTES: Long = 60 * 24 // Once every day
+    private val TAG: String = BackgroundTaskConsumer::class.java.simpleName
+  }
+
+  private var mTask: TaskInterface? = null
+  private var mBackgrounded: Boolean = true
+  private val taskCoroutineScope = CoroutineScope(Dispatchers.Default)
+
+  override fun taskType(): String {
+    return BACKGROUND_TASK_TYPE
+  }
+
+  /**
+   * Exposing the execute task function so that the BackgroundTaskWork (WorkManager Work unit)
+   * can execute the inner task. A task is only executed if the app is in the background.
+   * This only applies when the app is in release mode.
+   */
+  fun executeTask(callback: TaskExecutionCallback) {
+    if (mBackgrounded || ReactBuildConfig.DEBUG) {
+      Log.i(TAG, "Executing task '${mTask?.name}'")
+      taskManagerUtils.executeTask(mTask, null, callback)
+    } else {
+      Log.w(TAG, "Task was not executed since the app was not in the background or in Debug mode.")
```

Comment:
The idea is to try to avoid calling the task when in the background, since it is asynchronous we might get into situations where a task is scheduled when in the background - but run when in the foreground. It should not fail, just warn. But maybe it is better to just skip logging - I think I added the log to test when developing. :) :)